{
  Failure fail=new Failure(){
    @Override public void eval(    MockDirectoryWrapper dir) throws IOException {
      for (      StackTraceElement e : Thread.currentThread().getStackTrace()) {
        if (doFail && "createOutput".equals(e.getMethodName())) {
          throw new FakeIOException();
        }
      }
    }
  }
;
  MockDirectoryWrapper dir=newMockDirectory();
  dir.failOn(fail);
  Codec codec=getCodec();
  byte id[]=StringHelper.randomId();
  SegmentInfo info=new SegmentInfo(dir,getVersions()[0],"_123",1,false,codec,Collections.<String,String>emptyMap(),id,new HashMap<>());
  info.setFiles(Collections.<String>emptySet());
  fail.setDoFail();
  try {
    codec.segmentInfoFormat().write(dir,info,IOContext.DEFAULT);
    fail("didn't get expected exception");
  }
 catch (  FakeIOException expected) {
  }
 finally {
    fail.clearDoFail();
  }
  dir.close();
}
